### ISSUE TITLE
Development API key for dropbox plugin has expired

### ISSUE THREAD

**Issue by akloeber** [syncany/syncany#480]
When trying to authenticate syncany in Dropbox the following error is shown:
*"This app is in development mode and can't accept more users. Contact the app developer and ask them to use the Dropbox API App Console to apply for production status."*

__Environment:__
Syncany 0.4.5-alpha
dropbox plugin 0.4.5-alpha
Mac OS X 10.10.3

**Comment by cr0** [comment]
I am aware and I already applied for production a few days ago. However we have to wait for a response. In the meantime, you can create your own api key and compile the plugin with your own key. It'll work then... Sorry about that!

**Comment by markx7** [comment]
Any news and/or update?

**Comment by cr0** [comment]
We were rejected in the first try. The Dropbox team ran into some kind of error which caused the rejection. We are discussing possible error causes internally at the moment. We will reapply for production as soon as we have a clue what went wrong. Sorry.

**Comment by markx7** [comment]
thanks for your reply (and of course for the development of this great software!), I'll continue to follow you waiting for news

**Comment by cr0** [comment]
@binwiederhier: have you done any testing regarding this issue (see internal mail)

**Comment by binwiederhier** [comment]
I have not yet. Will do tonight.

**Comment by cr0** [comment]
Great!

**Comment by HubertMa** [comment]
The error "This app is in development mode and can't accept more users. Contact the app developer and ask them to use the Dropbox API App Console to apply for production status." still exists.

Is there some documentation about compiling the Dropbox plugin with an own key
as proposed by cr0? (I'm an experienced Linux user, but this description is too little for me.)
Or will there be a transition to "production mode" soon?

**Comment by pimotte** [comment]
The transition to production mode has hit a couple of bumps. 

You can compile it with your own key in the following way:

```
git clone git@github.com:syncany/syncany-plugin-dropbox.git
edit src/main/java/org/syncany/plugins/dropbox/DropboxTransferPlugin.java to include a key you requested from Dropbox
./gradlew pluginJar
syncany plugin install build/libs/*.jar
```

**Comment by HubertMa** [comment]
> The transition to production mode has hit a couple of bumps.
>
> You can compile it with your own key in the following way:
>
> |git clone git@github.com:syncany/syncany-plugin-dropbox.git
> edit src/main/java/org/syncany/plugins/dropbox/DropboxTransferPlugin.java to include a key you requested from Dropbox
> ../gradlew pluginJar
> syncany plugin install build/libs/*.jar
> |
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/syncany/syncany/issues/480#issuecomment-139823286>.
>
Thank you Pim, I think I'll try this.
Hubert

**Comment by kingjr** [comment]
> ```
git clone git@github.com:syncany/syncany-plugin-dropbox.git
edit src/main/java/org/syncany/plugins/dropbox/DropboxTransferPlugin.java to include a key you requested from Dropbox
./gradlew pluginJar
syncany plugin install build/libs/*.jar
```

After modifying the `APP_KEY` and `APP_SECRET`, compiling and installing the plug in, adding a folder up to 'Authorize', it launches the browser and dropbox throws this error:

```
Error (400)
It seems the app you were using submitted a bad request. If you would like to report this error to the app's developer, include the information below.


More details for developers

Invalid redirect_uri: "http://localhost:6462/dropbox/": It must exactly match one of the redirect URIs you've pre-configured for your app (including the path).

```

Have I missed something?

**Comment by cr0** [comment]
You need to add `localhost:6462` as a redirect URL in your Dropbox Developer Dashboard for your Syncany App.

**Comment by kingjr** [comment]
indeed, thanks @cr0

**Comment by Avamander** [comment]
I'm unable to recompile the plugin because of some weird configuration issue. What's the estimate of getting Dropbox's approval?

**Comment by jaysonsantos** [comment]
Hi, is it being ignored by Dropbox team?

**Comment by pimotte** [comment]
Not really. We got a stacktrace from them that we couldn't reproduce and as I recall we never followed up again.

**Comment by jaysonsantos** [comment]
Would be nice to have the tokens configurable in a config file or in the GUI with that you could give up on waiting for them to approve

**Comment by Avamander** [comment]
You should try and follow up again.
